src/main/java/org/takes/facets/hamcrest/AbstractHmTextBody.java

```diff
@@ -74,18 +73,25 @@ public AbstractHmTextBody(final Matcher<String> body,
 
     @Override
     public final void describeTo(final Description description) {
-        description.appendDescriptionOf(this.body);
+        description.appendText("body: ").appendDescriptionOf(this.body);
     }
 
     @Override
     protected final boolean matchesSafely(final T item) {
         try {
-            return this.body.matches(this.text(item));
+            this.itext = this.text(item);
```

Comment:
@olenagerasimova it's not correct usage of Hamcrest-matchers, it may cause inconsistent state of matcher. You can check correct value to describe in `describeMismatchSafely` argument.